### PR TITLE
Installing volsync using kustomize

### PR DESCRIPTION
**Pull request by larsks**
Now that our very exciting call is over, install volsync using our standard practice of exploding the helm chart into a set of kustomize manifests.

### CODE REVIEW

File: argocd/overlays/moc-infra/applications/envs/moc/smaug/cluster-management/volsync.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+metadata:
+  name: volsync
+  namespace: argocd
+spec:
+  project: cluster-management
+  source:
+    path: volsync/overlays/moc/smaug
+    repoURL: https://github.com/operate-first/apps.git
+    targetRevision: HEAD
+  destination:
```
Comment (HumairAK) [inline argocd/overlays/moc-infra/applications/envs/moc/smaug/cluster-management/volsync.yaml:12]: can we turn on auto sync? i.e.: 

```
  syncPolicy:
    automated:
      prune: true
      selfHeal: true
    syncOptions:
      - Validate=false
```

**Comment by HumairAK** [comment]
other than above lgtm

File: argocd/overlays/moc-infra/applications/envs/moc/smaug/cluster-management/volsync.yaml (hunk 1 above)
Comment (larsks) [reply argocd/overlays/moc-infra/applications/envs/moc/smaug/cluster-management/volsync.yaml:12]: Oops, yeah. All set.

File: argocd/overlays/moc-infra/applications/envs/moc/smaug/cluster-management/volsync.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    name: smaug
+    namespace: volsync-system
```
Comment (HumairAK) [inline argocd/overlays/moc-infra/applications/envs/moc/smaug/cluster-management/volsync.yaml:14]: we should also add this namespace declaratively in cluster-scope/base

**Comment by larsks** [comment]
(Your comment re: namespaces reminded me to add the namespace to the resource bundle.)

File: cluster-scope/bundles/volsync/kustomization.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+  - ../../base/core/namespaces/volsync-system
+  - ../../base/apiextensions.k8s.io/customresourcedefinitions/replicationdestinations.volsync.backube
+  - ../../base/apiextensions.k8s.io/customresourcedefinitions/replicationsources.volsync.backube
+  - ../../base/rbac.authorization.k8s.io/clusterrolebindings/volsync-manager
+  - ../../base/rbac.authorization.k8s.io/clusterrolebindings/volsync-proxy
+  - ../../base/rbac.authorization.k8s.io/clusterroles/volsync-manager
+  - ../../base/rbac.authorization.k8s.io/clusterroles/volsync-metrics-reader
```
Comment (HumairAK) [inline cluster-scope/bundles/volsync/kustomization.yaml:10]: is the scc being imported else where? (I don't see it)

Also what about the namespace, is that something that would go in the bundle (I would think so)?

Comment (larsks) [reply cluster-scope/bundles/volsync/kustomization.yaml:10]: > is the scc being imported else where? (I don't see it)

Whoops, yeah, missed that.

> Also what about the namespace, is that something that would go in the bundle (I would think so)?

Yeah, added that while you were reviewing :)

Comment (larsks) [reply cluster-scope/bundles/volsync/kustomization.yaml:10]: Updated.

**Pull request merged**